File: examples/example2.cpp
//=================================================================================================
//                    Copyright (C) 2016 Olivier Mallet - All Rights Reserved                      
//=================================================================================================

#include "../include/URT.hpp"

int main()
{
   int nobs = 1000;

   // generating non-stationary random data
   urt::Vector<double> data = urt::wiener_process<double>(nobs);

   // initializing ADF test with 10 lags and constant trend
   urt::ADF<double> test(data, 10, "ct");

   // outputting test results
   test.show();
    
   // switching to test with lag length optimization and p-value computation by bootstrap with 10000 iterations
   test.method = "AIC";
   test.bootstrap = true;
   test.niter = 10000;
    
   // outputting test results
   test.show();  
}

File: R/example2.R
#==================================================================================================
#                     Copyright (C) 2016 Olivier Mallet - All Rights Reserved                      
#==================================================================================================

suppressMessages(library(RcppURT))

run <- function()
{
  x = as.matrix(read.table("../data/x.csv", sep=","))
  y = as.matrix(read.table("../data/y.csv", sep=","))

  # running OLS regression as in ./examples/example1.cpp using double precision type
  fit = OLSreg_d(y, x, stats=TRUE, output=TRUE)

  # running OLS regression as in ./examples/example1.cpp using single precision type
  fit = OLSreg_f(y, x, stats=TRUE, output=TRUE)

  # running first ADF test as in ./examples/example2.cpp using double precision type
  test = ADFtest_d(y, lags=10, trend='ct', output=TRUE)

  # running second ADF test as in ./examples/example2.cpp using double precision type
  test = ADFtest_d(y, method='AIC', trend='ct', output=TRUE, bootstrap=TRUE, niter=10000)
}
